www/localPackage.ts

```diff
@@ -118,13 +117,18 @@ class LocalPackage extends Package implements ILocalPackage {
                     } else {
                         var invokeSuccessAndInstall = () => {
                             CodePushUtil.logMessage("Install succeeded.");
-                            installSuccess && installSuccess();
-                            /* no neeed for callbacks, the javascript context will reload */
-                            cordova.exec(() => { }, () => { }, "CodePush", "install", [deployDir.fullPath, installOptions.rollbackTimeout.toString(), installOptions.installMode.toString()]);
+                            if (installOptions.installMode === InstallMode.IMMEDIATE) {
+                                /* invoke success before navigating */
+                                installSuccess && installSuccess();
+                                /* no neeed for callbacks, the javascript context will reload */
+                                cordova.exec(() => { }, () => { }, "CodePush", "install", [deployDir.fullPath, installOptions.installMode.toString()]);
+                            } else {
+                                cordova.exec(() => { installSuccess && installSuccess(); }, () => { installError && installError(); }, "CodePush", "install", [deployDir.fullPath, installOptions.installMode.toString()]);
+                            }
                         };
 
                         var preInstallSuccess = () => {
-                            Sdk.reportStatus(AcquisitionStatus.DeploymentSucceeded);
+                            Sdk.reportStatus(AcquisitionStatus.DeploymentSucceeded, this);
```

Comment:
I changed part of the reporting mechanism to be initiated on the native side. This is how it currently works:
- For install failures (unzip failed, start page not found, etc) we initiate reporting in the JS layer.
- For rollbacks, install confirmations and store updates, we initiate it from the native layer. Moreover, when any of these events happen, I cache the event on the native side, and only send it over to the JS layer when the application is paused (loses focus). This way, the user experience is not degraded by telemetry.

One thing I wanted to talk about is the store update scenario. In the current implementation, I send a telemetry report for all store updates but not the initial install. Do we need an event for the first (clean) install as well?